developer/Dockerfile

```diff
@@ -1,22 +1,23 @@
-ARG IMG=jammy
+ARG IMG=noble    # Ubuntu image that contains all corresponding dependencies.
 
 FROM dealii/dependencies:$IMG
 
-LABEL maintainer="luca.heltai@gmail.com"
-
-ARG VER=master
-ARG PROCS=2
+ARG NJOBS=0      # Jobs used for building. Default: Use all available jobs.
+ARG VER=master   # deal.II branch that gets checked out.
+ARG FLAGS=""     # Additional flags for the build.
 
 USER root
-RUN cd /usr/src \
-    && git clone https://github.com/dealii/dealii dealii-$VER \
-    && cd dealii-$VER && git checkout $VER && \
-    mkdir build && cd build \
-    && cmake -GNinja \
+
+RUN cd /usr/src && \
+    git clone https://github.com/dealii/dealii dealii-$VER && \
+    cd dealii-$VER && \
+    git checkout $VER && \
+    mkdir build && cd build && \
+    cmake -GNinja \
     -DCMAKE_PREFIX_PATH="/usr/lib/x86_64-linux-gnu/hdf5/openmpi;/usr/include/hdf5/openmpi" \
     -DDEAL_II_ALLOW_AUTODETECTION=OFF \
     -DDEAL_II_COMPONENT_PYTHON_BINDINGS=ON \
-    -DCMAKE_CXX_FLAGS="-std=c++17" \
+    -DCMAKE_CXX_FLAGS="-std=c++20 $FLAGS" \
```

Comment:
Do we need  `-std=c++20` here? Same points that were made about the `-std=c++17` by @tamiko